### PR TITLE
Atom/mriegger/asvmsaa2x

### PR DESCRIPTION
**Pull request by mrieggeramzn**
Switch ASV to use 2x msaa

### CODE REVIEW

File: Gem/Code/Source/SampleComponentManager.cpp [hunk 1]
```diff
@@ -1634,4 +1632,20 @@ namespace AtomSampleViewer
...
+        AZ_Assert(IsValidNumMSAASamples(m_numMSAASamples), "Invalid MSAA sample setting");
+        const AZ::RHI::SamplePosition Center = { 8, 8 };
+        const AZ::RHI::SamplePosition TopLeft = { 0, 0 };
+
+        AZ::RHI::MultisampleState state;
+
+        state.m_samples = static_cast<uint16_t>(m_numMSAASamples);
+        state.m_customPositionsCount = 2;
+        state.m_customPositions[0] = Center; 
+        state.m_customPositions[1] = TopLeft;
```
Comment (antonmic) [inline Gem/Code/Source/SampleComponentManager.cpp:1646]: Can you fill in all 16 positions with the list I gave you? That way someone can easily switch MSAA back to a different count and all will still work.

**Review by antonmic — CHANGES_REQUESTED** [review]

File: Gem/Code/Source/SampleComponentManager.cpp (hunk 1 above)
Comment (mrieggeramzn) [reply Gem/Code/Source/SampleComponentManager.cpp]: Done

File: Gem/Code/Source/SampleComponentManager.cpp [hunk 2]
```diff
@@ -1634,4 +1632,31 @@ namespace AtomSampleViewer
...
     }
 
+    AZ::RHI::MultisampleState SampleComponentManager::CreateDefaultMsaaState() const
+    {
+        AZ_Assert(IsValidNumMSAASamples(m_numMSAASamples), "Invalid MSAA sample setting");
+
+        AZ::RHI::MultisampleState state;
+        state.m_samples = static_cast<uint16_t>(m_numMSAASamples);
+        state.m_customPositionsCount = m_numMSAASamples;
+        state.m_customPositions[0] = { 8, 8 };
```
Comment (moudgils) [inline Gem/Code/Source/SampleComponentManager.cpp:1642]: Add a comment about the custom positions. Would be nice to see a diagram of it.

Comment (mrieggeramzn) [reply Gem/Code/Source/SampleComponentManager.cpp:1648]: Added a comment. Good idea since our notation is different from hlsl

Comment (moudgils) [reply Gem/Code/Source/SampleComponentManager.cpp:1648]: Nit. Maybe add a line saying 

"We are filling all the possible positions here even though the positions taken into consideration will be capped by m_customPositionsCount"

**Review by moudgils — APPROVED** [review]

**Comment by moudgils** [comment]
Nit. Maybe add a line saying

"We are filling all the possible positions here even though the positions taken into consideration will be capped by m_customPositionsCount"

**Review by antonmic — APPROVED** [review]

**Pull request merged**